pkg/obfuscate/sql_tokenizer.go

```diff
@@ -276,8 +276,18 @@ func (tkn *SQLTokenizer) Scan() (TokenKind, []byte) {
 				return TokenKind(ch), tkn.bytes()
 			}
 		case '#':
-			tkn.advance()
-			return tkn.scanCommentType1("#")
+			// hack: SQL Server supports the "#" syntax for temp tables, but MySQL treats
```

Comment:
I don't know about this. It scares me. What if we break someone? 😕 I was wondering if we wouldn't be able to add an extra span to tags, which specify the SQL engine. We could use it here and have the client send the tag `sql.enging:"sql-server"` and identify the correct behaviour. Would that work for your use case? Would you be able to use this tag? I'm not sure how you are using the obfuscator.

Reply:
I don't think span tags would solve our particular use case, but we can take inspiration off that. I think we can pass the engine type through our [SQL options](https://github.com/DataDog/datadog-agent/blob/8d4a5574ea4a542f3a024b4dc97becab5bef2a3c/pkg/collector/python/datadog_agent.go#L224-L236) via the Python -> Go stub. This would allow us to get this context[ inside the SQL tokenzier](https://github.com/DataDog/datadog-agent/blob/1b40cc12b62d0f6f447fa53081ce2f85078976e6/pkg/obfuscate/sql_tokenizer.go#L174) without much effort.

Something like:
```
type sqlConfig struct {
    EngineType string `json:"engine_type"`
    .... other options
}
```
And because this config will be inside the tokenzier we can just do a simple check.
`case ' ' && tkn.cfg.EngineType == "sqlserver"`